[Nobel Ang] iP

codecodecoder78 opened the pull request:
# iP pull request

These are the included changes
* Commands are contained in a **Enum**
* Input and outputs are contained in Ui class
* Parsing information included in Parser class
* *Task* is abstract and the 3 task types inhereits from Task

The current Task Classes
1. Todo
2. Deadline
3. Event

```java
function fancyAlert(arg) {
  if(arg) {
    $.facebox({div:'#foo'})
  }
}
```

ToDo
- [x] More OO
- [x] ~~Styling and Git~~
- [ ] Week 4 iP :exploding_head:

> A 4 mc mod
> should only take 10 hours a week `return`

https://www.youtube.com/watch?v=dQw4w9WgXcQ

Commit:
Add Gradle support

codecodecoder78 committed:
Level 1: Implemet echo function

codecodecoder78 committed:
Level-1: Fix Bye printing

codecodecoder78 committed:
Level 1: Fix bye casing

codecodecoder78 committed:
Level 2: Implement List

codecodecoder78 committed:
Level 3: Add tasks

codecodecoder78 committed:
Level 4: Support for 3 types of task

codecodecoder78 committed:
A-TextUiTesting: Set up testing

codecodecoder78 committed:
Level-5: Handle Errors

codecodecoder78 committed:
Level-6: Delete

codecodecoder78 committed:
Add save and load feature and Javadocs

List of tasks will be saved after every change and restored on new load.

Javadocs added for all public method and classes.

codecodecoder78 committed:
Change /by format for deadline to LocalDate

/by format for deadline is stored in LocalDate.
 /by inputs will have to be in yyyy-mm-dd (eg. 2021-02-01).

codecodecoder78 committed:
Merge branch 'branch-Level-7'

codecodecoder78 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

codecodecoder78 committed:
Seperated interactions into seperate classes

Ui class contains methods for receiving user input and printing user output.
Command enum tracks expected commands.
Storage class handles saving and loading.
Parser handles parsing of commands.

codecodecoder78 committed:
Add all files to duke package

codecodecoder78 committed:
Add test for ToDo and TaskList

codecodecoder78 committed:
Add JavaDoc comments

codecodecoder78 committed:
Fix style error

codecodecoder78 committed:
Add find command

codecodecoder78 committed:
Merge branch 'branch-A-JavaDoc'

codecodecoder78 committed:
Merge branch 'branch-A-CodingStandard'

codecodecoder78 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

LZher385 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,82 @@
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String nextLine = s.nextLine();
+                String[] nextItem = nextLine.split("\\Q|\\E", 0);
```
Nice use of regex 👍

LZher385 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,116 @@
+     * @return Array of indexes.
+     */
+    public ArrayList<Integer> find(String query) {
+        ArrayList<Integer> output = new ArrayList<>();
```
Shouldn't the name be in plural form?

LZher385 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,60 @@
+ */
+public class Event extends Task {
+
+    protected String at;
```
I think the variable name could be a little more descriptive.

codecodecoder78 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

codecodecoder78 committed:
Fix coding standard issues

codecodecoder78 committed:
Merge branch 'branch-A-Gradle'

didymusne commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+package duke; /**
```
Should the comment start from line 2?

didymusne commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,126 @@
+     * @throws DukeException When TaskList cannot retreive the last element in the list.
+     */
+    public void printNewTask(TaskList tasks) throws DukeException {
+        System.out.println("---------");
```
I think this "-----", "-----", can be abstracted as a method, e.g. formatMessage()

didymusne commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,72 @@
+    @Override
+    public String toString() {
+        String output = "";
+        if (this.getStatus()) {
```
I think this can be simplified into one return statement!

E.g. return this.getStatus()
             ? "[x] " + this.getDescription()
              : "[ ] " + this.getDescription()

didymusne commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,135 @@
+            return new String[] {input.split(" ")[1].trim()};
+        case FIND:
+            return new String[] {input.replace("find ", "").trim()};
+        default :
```
I think there does can be no white space after "default"

didymusne commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,135 @@
+     * @param array Array to be trimmed.
+     */
+    private void trimMap (String[] array) {
+        for (int i = 0 ; i < array.length; i++) {
```
I think this can be simplified with the `map(x -> f(x))` function!

didymusne reviewed (approved):
LGTM! I enjoyed reading this code and it is very tidy, except for very minor comments! Good job!

codecodecoder78 committed:
Add GUI
Added chat style GUI.
Print functions return the string printed as a output.

codecodecoder78 committed:
Merge branch 'branch-Level-10'

codecodecoder78 committed:
Add assert statment to ensure that the program exits when a Bye Command is issued

codecodecoder78 committed:
Add default to switch statements

codecodecoder78 committed:
Merge pull request #1 from angnobel/branch-A-Assertions

A-Assertions

codecodecoder78 committed:
Merge branch 'master' of https://github.com/angnobel/ip

codecodecoder78 committed:
Merge branch 'master' into branch-A-CodeQuality

codecodecoder78 committed:
Add default case to Switch statments

There are 2 switch statements that is used to process Commands. These statements did not have default cases, which may cause error if an unrecognised command was passed in.

Adding default statements improves code quality and error checking.

codecodecoder78 committed:
Merge branch 'master' of https://github.com/angnobel/ip

* 'master' of https://github.com/angnobel/ip:
  Add default to switch statements

codecodecoder78 committed:
Add remind function to Duke

codecodecoder78 committed:
Merge branch 'branch-B-Reminder'

codecodecoder78 committed:
Add user guide

codecodecoder78 committed:
Correct build.gradle

codecodecoder78 committed:
Add find command to documentation
